topic_dynamics/parsing.py

```diff
@@ -0,0 +1,150 @@
+    return list_of_files
+
+
+def read_file(file: str) -> bytes:
+    """
+    Read the contents of the file.
+    :param file: address of the file.
+    :return: bytes with the contents of the file.
+    """
+    with open(file, 'r') as fin:
+        code = bytes(fin.read(), 'utf-8')
+    return code
+
+
+def get_positional_bytes(node: tree_sitter.Node) -> Tuple[int, int]:
+    """
+    Extract start and end byte.
+    :param node: node on the AST.
+    :return: (start byte, end byte)
+    """
+    start = node.start_byte
+    end = node.end_byte
+    return start, end
+
+
+def get_identifiers(file: str, lang: str) -> List[Tuple[str, int]]:
+    """
+    Gather a sorted list of identifiers in the file and their count.
+    :param file: address of the file.
+    :param lang: the language of file.
+    :return: a list of tuples, identifier and count.
+    """
+    code = read_file(file)
+    tree = get_parser(lang).parse(code)
+    root = tree.root_node
+    identifiers = []
+    node_types = {'c': ['identifier', 'type_identifier'],
+                  'c-sharp': ['identifier', 'type_identifier'],
+                  'cpp': ['identifier', 'type_identifier'],
+                  'java': ['identifier', 'type_identifier'],
+                  'python': ['identifier', 'type_identifier']}
+
+    def traverse_tree(node: tree_sitter.Node) -> None:
+        """
+        Run down the AST from a given node and gather identifiers from its childern.
+        :param node: starting node.
+        :return: None.
+        """
+        for child in node.children:
+            if child.type in node_types[lang]:
+                start, end = get_positional_bytes(child)
+                identifier = code[start:end].decode('utf-8')
+                if '\n' not in identifier:  # Will break output files. Can add other bad characters later
+                    identifiers.append(identifier)
+            if len(child.children) != 0:
+                traverse_tree(child)
+
+    traverse_tree(root)
+    sorted_identifiers = sorted(Counter(identifiers).items(), key=itemgetter(1), reverse=True)
+
+    return sorted_identifiers
+
+
+def transform_identifiers(identifiers: List) -> List[str]:
+    """
+    Transform the original list of identifiers into the writable form.
+    :param identifiers: list of tuples, identifier and count.
+    :return: a list of identifiers in the writable for, "identifier:count".
+    """
+    formatted_identifiers = []
+    for identifier in identifiers:
+        if identifier[0].rstrip() != '':  # Checking for occurring empty tokens.
+            formatted_identifiers.append(identifier[0].rstrip() + ':' + str(identifier[1]).rstrip())
+    return formatted_identifiers
+
+
+def main(repository: str, number: int, delta: int, lang: str, output: str, output_info: str) -> None:
+    """
+    Split the repository, parse the files, write the data into a file.
+    :param repository: path to the repository to process, must have a git file.
```

Comment:
```suggestion
    :param repository: path to the repository to process, must be a git repository.
```
If it must be a git repository - check it - `assert os.path.isdir(os.path.join(repository, ".git"))`

Reply:
Will do it in the next PR, because it changes the main function.

Reply:
Will ask you more about asserts.